Ported in a variety of gameplay actions, quick bug fixes

Nobonet opened the pull request:
kept the idea of making a pull request in mind, it's not exactly everything since im trying to keep it compatible with the current structure of the repo itself
_(Could this mean that someone has to take care of the missing animations .. ?)_

- added
  - quicksand behavior (HasTag "Quicksand" OR part.name matches "quicksand") (QuicksandType attribute to specify)
  - hangable ceil climbing (HasTag "Hangable" OR Material == DiamondPlate (?))
  - updating health, fall damage actions applied
  - getting stuck on snow/sand ground

Nobonet committed:
Update README.md

Nobonet committed:
Update README.md

Nobonet committed:
Add missing return m.SetAction for checkWallkick

Previously did not have it, causing for more precise wall kicks and less time to perform one

Nobonet committed:
Update README.md

Nobonet committed:
wrong thing

Nobonet committed:
Update README.md

lets use vinesauce video clip instead :D

Nobonet committed:
Add UpdateHealth

Nobonet committed:
Quicksand stuffs

Might not work yet

Nobonet committed:
Max isn't using snake_case

lol

Nobonet committed:
Add cancels mess

Nobonet committed:
Update init.lua

Nobonet committed:
Some quicksand actions

Nobonet committed:
Add the rest of quicksand behavior

Nobonet committed:
You better know this

Nobonet committed:
Also know all of this

Nobonet committed:
Some quicksand actions

Nobonet committed:
Info

jolurumajrjr-blip committed:
super evil mode update

GREAT!

jolurumajrjr-blip committed:
info

jolurumajrjr-blip committed:
comments and comments and statements

jolurumajrjr-blip committed:
added hangable ceil

jolurumajrjr-blip committed:
missing states and its checks and more junk

- added health net dispatching
- added some missing death actions and its statement checking
- water knockback stuff
- minor value changes

jolurumajrjr-blip committed:
change some values back

jolurumajrjr-blip committed:
just a few fixes and additions

- attempt to fix ledge climbing down
- wrong function used in hangable ceil animation
- safety check????
- allow punching off a platform
- change panting animation index

jolurumajrjr-blip committed:
forgot to use m.ActionArg for hangable ceil anim..

jolurumajrjr-blip committed:
pull request remove that

i might make a requiest to put this all onto the original repo

jolurumajrjr-blip committed:
organize and tidy up

remove and sort and relocate and have fun

jolurumajrjr-blip committed:
brief statements and accomodations

- reset quicksanddepth at onReset()
- comment something out LOL
- specify quicksand type using attributes

jolurumajrjr-blip committed:
Fix hangable ceil wrong behavior

Did not check if the difference is out of (-30, 30), so you'll get upwarped moving off a hangable ceil and above you is an even higher hangable ceil

jolurumajrjr-blip committed:
Ignore TAS input on hangable ceiling movement

otherwise you'd keep moving

jolurumajrjr-blip committed:
add more to TAS input exclusions

- ignore TAS on hangable ceiling climbing
- ignore TAS on air states (B button will sink when holding A, did this just so you can do jumpkicks easily and while holding the A button)

jolurumajrjr-blip committed:
Respect typechecking

jolurumajrjr-blip committed:
tas input ignore fix

didnt exclude for wallkicks LOLOLOL

jolurumajrjr-blip committed:
tas input ignore better solution for real this time

Great job!

jolurumajrjr-blip committed:
Update init.server.lua

jolurumajrjr-blip committed:
Physics additions and states

- Add squishing
- Add inertia (toggleable, FFLAG_USE_INERTIA) Uses part Velocity and Angular Velocity
- add health to debug:)

jolurumajrjr-blip committed:
Strict squishing ceilings to tagged objects

tag a part with "SquishMario" or use unanchored parts for Mario to get squished under them

jolurumajrjr-blip committed:
suiqshtimers and debugs and unimportancy

jolurumajrjr-blip committed:
expanding to FloorType, extra binds

- use GetFloorType on some scenarios
- super tuper duper secret hold-bind to turn off tas input (i will Now jumpkick up all of your slopes .)

jolurumajrjr-blip committed:
Collision faces debug (photosens warning)

Photosensitivity warning for this feature!! Sometimes causes flashing colors

includes cheat check for transparency 1 geometry for non-studio sessions

jolurumajrjr-blip committed:
ignorable collision types

you may ignore Wall, Ceil or Floor by:
- adding a tag to a part named CollIgnore{type}, such as CollIgnoreWall

use for BLJ staircases

jolurumajrjr-blip committed:
Luau typechecking troubles

super duper fun adventure of happy time

jolurumajrjr-blip committed:
untruncate ceil height when needed to

jolurumajrjr-blip committed:
small code refactors

jolurumajrjr-blip committed:
weird crap going on here

jolurumajrjr-blip committed:
update fn Mario.HandleSpecialFloors

- Fixed bug where you still get burnt by lava floors when being above them via hangable ceiling

jolurumajrjr-blip committed:
offest camera with gfxpos, ceilheight stuff

jolurumajrjr-blip committed:
tas input override changes

top secret confidential information let me jumpkick up all your slopes

jolurumajrjr-blip committed:
State additions, some of them yet to be fully implemented

- Reset loaded animations table on CharacterAdded (if you hotswap the Character yourself or sudden respawn time)
- Add hurt sound to knockback
- Update squish behavior
- Add Mario.GfxScale (You have to use this manually for transforming your model!)
- Add FIRST_PERSON behavior (currently unavailable by normal means)
- Extend Floor raycast
- Add a single missing sound

jolurumajrjr-blip committed:
shoutouts

to simpleflips

- get surface type function changes
- ray extension fix
- make lazy network remote into unreliable

ghost commented:
nobonet touch scripts wont work lets just make it work again?

jolurumajrjr-blip committed:
easiest fix of my stupid chud life

Ok. take this. Couldnt punch off ledges nor get pushed off by conveyors right . Thats cuz max forgot to do m.Position = nextPos in the qstep solver. Lol. That's pretty silly. OOPS!

also reverted the getsurfacetype function idk i didnt like it

MaximumADHD commented:
This all looks pretty good, might actually merge it. I'll get back to you.

jolurumajrjr-blip committed:
add windy winds

- include HORIZONTAL_WIND and VERTICAL_WIND (i guess) and its cohorts

soundtable existingsound check (report performance later?)

jolurumajrjr-blip committed:
misinput

forgot to apply platform inertia somewhere

jolurumajrjr-blip committed:
mistakes and flags

- SignedShort is in fact an int16 method
- add fflag FFLAG_SGS_ALWAYS_PERFORMS_STEPS (Whether you would want the stationarygroundstep to cast wall collisions by calling PerformGroundStep all times or not)

jolurumajrjr-blip committed:
the quest for 5 million dollars

find floor extend

Nobonet committed:
Update .gitattributes

jolurumajrjr-blip committed:
Redo util (AGAIN)

jolurumajrjr-blip committed:
untested code alert

turns out rojo wasnt connected and on the accept prompt so i thought everything worked LOLOLOLOLOLOLOL

SORRY

jolurumajrjr-blip committed:
Water and Gas cloud planes

- Added gas clouds
- To define a gas cloud plane, give a part the tag "PoisonGasCloud" and make sure it's CanCollide false
- To define a water plane, do the same thing noted above but the tag is instead just "Water"
- typechecking fixes
- reset inertia on bonk

jolurumajrjr-blip committed:
Action.COUGHING duplicate

sorry

Nobonet committed:
Add files via upload

jolurumajrjr-blip committed:
bugfixes

- append tag params better
- surfaceclass extra checks

jolurumajrjr-blip committed:
Merge branch 'main' of https://github.com/Nobonet/sm64-roblox

jolurumajrjr-blip committed:
code acommodations, new gameplay additions

- Added shell riding actions
- Commented SurfaceClass with descriptions
- Added the Interaction module (SM64.Client.Game.Interaction)
-- Allows you to cast damage, set shell riding and cast flame to Mario (TODO add more actions?)

jolurumajrjr-blip committed:
Update sm64-roblox

just a whole lot of patches, replacements and redos

jolurumajrjr-blip committed:
Add Interaction.InteractCap, plus other stuff

- added Interaction.InteractCap(m: Mario, capFlag: number)
- added fflag FFLAG_AUTO_RESET_ON_DEAD (located on SM64.Client's code)
- type stuff

jolurumajrjr-blip committed:
getwalltype

the birth of a new function the start of a new life

jolurumajrjr-blip committed:
Oops

TYPO

jolurumajrjr-blip committed:
Add star behavior, minor nitpicks fixed, etc

- Added star dance actions and Interaction.InteractStarOrKey(m: Mario)
- Added DEBUG_FREE_MOVE (trigger manually)
- tiny fixes

jolurumajrjr-blip committed:
Value changes/additions

- changed auto reset to true
- don't set faceangle on stardance multiple times
- added stuff to InteractionType enum (later use??)

jolurumajrjr-blip committed:
Several additions & changes

- Added climbing poles (e.g. trees), use Interaction.InteractPole(m: Mario, pole: BasePart)
- AssemblyVelocity inertia should be just accurate enough
- Don't overlap Mario sounds (added priority system)
- Added SM64 object types & enums (Comes without a solution for SM64 objects, you may have to do these yourself, might think about releasing my own solution later?)
- Code refactoring / moving around
- Fixed mistakes (fall damage should get ignored by wing cap)
- Moved GetFloorType, GetCeilType, GetWallType to Util (Mario has redirects to these)
- Added water currents (Attribute "Force", of surfaceclass FLOWING_WATER)
- Added missing sound MARIO_EEUH (Slow ledge climb, fixed wrong long jump land sound as well)

jolurumajrjr-blip committed:
[ACTION REQUIRED BY USER] Changed interaction to accomodate for Objects

CANNOT COME WITH BACKWARDS COMPATIBILITY, READ THE WIKI HERE FOR NEW USAGE: https://github.com/Nobonet/sm64-roblox/wiki/Interaction-Module/

- Added Object data support, pseudo-objects for emulating them
- Moved ProcessInteractions to Interaction (Mario has redirects to this, pretty bad fix done)
- Updated DebugDisplay.rbxm
- Minor fixes

jolurumajrjr-blip committed:
Behavior additions & fixes

- Added Inertia displacement to hangable ceilings
- Minor fixes on Interaction.lua
- Added new fflag bools
- Fix some crouching actions
- NOT_SLIPPERY should not make mario slide

jolurumajrjr-blip committed:
Butchered object stuff & new additions

- added (probably) all the HOLD_ actions
- more object related stuff of course I Plan to do something devious ...
- code behavior changes & fixes
- Added FFLAG_BETTER_HANGING (From HackerSM64)
- Code snippet location moving arounds

jolurumajrjr-blip committed:
Code update/fixes

- Move PerformHangingStep as a Mario metamethod
- Fixed Subtype enum (No hexadecimal indication)
- Fixed Interaction module crazy ass visualstudio code autocomplete snipe IM ANGRY UGH
- Added command-walking for Keyboard (hold LeftControl to walk)
- More object related stuff (grabbing in this case) fixes
- Code cleanup
-

jolurumajrjr-blip committed:
another fix-bundled bundle

i forgot what i fixed

- Added spawn locations support (see FFLAG_USE_SPAWNLOCATIONS)
- MORE OBJECT RELATED FIXES LOLOLOLOLLOL
- moved inertia stuff to PlatformDisplacement
- fix wrong anim function on submerged
- add attribute CameraOffset (mario need property sorry. Have attribute as compensation)

jolurumajrjr-blip committed:
boy

- add fflag FFLAG_WALL_SLIDING (ported from sm64plus, adjusted a little bit)
- added absolutely worthless Camera enums (might or might not do something crazy with em later)
- code move arounds / sorting
- added hard surfaces (use tag "HardSurface")
- OOOOOOOBJEEEEEEEEEEEEEEECT FIXEEEEEEEEEES (new localization updates at this point)
- fixed stuck in ground incorrect action choosing

Nobonet committed:
wrong variable bug

OOOOOOOOPS

Nobonet committed:
fast turning flag

no more falling off platforms

- Added fflag FFLAG_FAST_TURNING (in Client.Mario.Moving), if youre about standing still you'll be slowed down and you'll turn faster if you're holding the stick away a bit

Nobonet committed:
merge fflags into one convenient module

SM64.Client.FFlags

i dont know if i missed anything but im pretty confident its everything

Nobonet committed:
boy

use real vanilla mode

Nobonet committed:
Don't loop AnimFrame

i've tested this with project64 + STROOP

- SHOCKED animation is meant to loop as well
- call onReset() at startup
- rename fflag DISPLACE_POS_Y to PLATFORM_DISPLACE_POS_Y
- register action WALL_SLIDE even if the fflag stated not enabled (not sure if anyone would change these values at runtime)
-  reset AnimFrame if AnimReset == true

Nobonet committed:
(likely) fix squish set action overflow crash

LOL

- moved common (x) cancels function outside

Nobonet committed:
additions & fixes

- fix actiongroup cancels wrong behavior (itwas just code max wrote which he wasnt expecting any of this new stuff to exist So it was my fault i overlooked it Sorry)
- add exiting states
- add fflags FALL_DAMAGE_HEIGHT and HARD_FALL_DAMAGE_HEIGHT (set to -1 to disable fall damage for which)

Gecmt commented:
Hey, i just wonder if you can help me. If it try to teleport the character with a script, it doesnt work. It stays in the same position. How can i fix it?

Nobonet committed:
dope mode activated

- Add boundary-related FFlags
- fix a function in Util
- Make assemblyvelocity-based platform inertia much more accurate (thx magicoal_nerb)
- variable & function renames/relocations
- adjust function for "continuous" collision (should detect slopes better)
- add a FFlag to enable a fake floor death plane (i thought i commited this?)
- shared sounds iterate through first
 instead of findfirstchild in __index

MarioGladiator commented:
@MaximumADHD Is it possible to make a ocarina of time sm64 roblox port it's been decompiled fully too https://github.com/zeldaret/oot

Nobonet committed:
Added Objects

Added an Object module to work with SM64's Object system, it comes with a Goomba object worker as an example of how it works.
:-)

real changes:
- Redid some Type definitions across sources
- Minor bug fixes
- Split PunchState to PunchType, PunchTimer (For no reason in particular)

There are no new additions... This repository has everything I think it needs to have.

Nobonet committed:
updated the localization files

- Changed type definitions for Mario
- Removed Alfred

Nobonet commented:
Thinking about it Perhaps it should be its own thing instead and that both repositories can coexist ? there's a whole lot of stuff that's new/changed and not sure if it's too much
